### PR TITLE
Write failed files (errors) to an errors.log and summarize to console

### PR DESCRIPTION
**Pull request by yma88**

### CODE REVIEW

File: charon/pkgs/maven.py [hunk 1]
```diff
@@ -13,30 +13,30 @@
...
-import sys
-import logging
-import re
-
+from charon.pkgs.pkg_utils import upload_post_process, rollback_post_process, log_errors
+from charon.storage import S3Client
+from charon.utils.archive import extract_zip_all
+from charon.utils.files import HashType
+from charon.utils.files import overwrite_file, digest
+from charon.utils.strings import remove_prefix
```
Comment (ligangty) [inline charon/pkgs/maven.py:39]: nitpick: are these importing changing really needed?

File: charon/storage.py [hunk 2]
```diff
@@ -13,19 +13,20 @@
...
 from botocore.exceptions import HTTPClientError
-from botocore.config import Config
-from boto3_type_annotations import s3
-from typing import Callable, Dict, List, Optional, Tuple
-import os
-import logging
-import mimetypes
+
+from charon.constants import PROD_INFO_SUFFIX
+from charon.utils.files import read_sha1
```
Comment (ligangty) [inline charon/storage.py:29]: nitpick: are these importing changing really needed?

**Review by ligangty — CHANGES_REQUESTED** [review]
@yma88 I think what we should do is not to record the errors twice, but just separate them to a specified error.log file. BTW, Instead of the way of a customized file writing, I think what we need is an extra logger with file appender. In this way we can:
* Avoid to return all errors back to upper caller.
* A unified way to configure the logger for future change. 

An addition, I think we should record all warning and errors here to this new log appender for tracking.

**Comment by yma88** [comment]
> @yma88 I think what we should do is not to record the errors twice, but just separate them to a specified error.log file. BTW, Instead of the way of a customized file writing, I think what we need is an extra logger with file appender. In this way we can:
> 
>     * Avoid to return all errors back to upper caller.
> 
>     * A unified way to configure the logger for future change.
> 
> 
> An addition, I think we should record all warning and errors here to this new log appender for tracking.

Hi,  @ligangty as I didn't see the logger file appender is required in the story, so just doing the customized way as one file plus error per line. I'm ok to adjust to the way of logger file appender, will update the story about this.

File: charon/storage.py (hunk 2 above)
Comment (yma88) [reply charon/storage.py:29]: @ligangty This is auto-format when I do the code formatting based on the PEP 8 style, if you have a preferred unified style setting of imports, welcome to share with me. Thanks.

**Pull request closed without merging**